cartography/intel/aws/cloudtrail_management_events.py

```diff
@@ -0,0 +1,553 @@
+
+    This creates the aggregated data needed for direct ASSUMED_ROLE relationships with
+    properties like times_used, first_seen, last_seen, and lastused.
+
+    :type role_assumptions: List[Dict[str, Any]]
+    :param role_assumptions: List of individual role assumption records
+    :rtype: List[Dict[str, Any]]
+    :return: List of aggregated role assumption relationships
+    """
+    aggregated = {}
+
+    for assumption in role_assumptions:
+        source_arn = assumption.get("SourcePrincipal")
+        dest_arn = assumption.get("DestinationPrincipal")
+        event_time = assumption.get("EventTime")
+
+        if not source_arn or not dest_arn or not event_time:
+            logger.warning(
+                "Skipping incomplete assumption record: due to missing required fields"
+            )
+            continue
+
+        # Convert STS assumed-role ARNs to IAM role ARNs for source matching
+        normalized_source_arn = _convert_assumed_role_arn_to_role_arn(source_arn)
+
+        # Create aggregation key
+        key = (normalized_source_arn, dest_arn)
+
+        if key not in aggregated:
+            aggregated[key] = {
+                "source_principal_arn": normalized_source_arn,
+                "destination_principal_arn": dest_arn,
+                "times_used": 1,
+                "first_seen": event_time,
+                "last_seen": event_time,
+            }
+        else:
+            # Update aggregated values
+            agg_data = aggregated[key]
+            agg_data["times_used"] += 1
+
+            # Update temporal bounds
+            if event_time < agg_data["first_seen"]:
+                agg_data["first_seen"] = event_time
+            if event_time > agg_data["last_seen"]:
+                agg_data["last_seen"] = event_time
+
+    # Convert to list and ensure lastused equals last_seen
+    result = []
+    for agg_data in aggregated.values():
+        agg_data["lastused"] = agg_data["last_seen"]
+        result.append(agg_data)
+
+    logger.info(
+        f"Aggregated {len(role_assumptions)} events into {len(result)} unique role assumption relationships"
+    )
+    return result
+
+
+def _extract_role_assumption_from_event(
+    event: Dict[str, Any],
+    region: str,
+    current_aws_account_id: str,
+) -> Optional[Dict[str, Any]]:
+    """
+    Extract role assumption details from a single CloudTrail event.
+
+    :type event: Dict[str, Any]
+    :param event: Single CloudTrail event from lookup_events API
+    :type region: str
+    :param region: The AWS region where the event occurred
+    :type current_aws_account_id: str
+    :param current_aws_account_id: The AWS account ID being synced
+    :rtype: Optional[Dict[str, Any]]
+    :return: Role assumption record or None if event is not a role assumption
+    """
+    event_name = event.get("EventName")
+
+    # Only process STS role assumption events
+    if event_name not in [
```

Comment:
See above comment about checking on if its possible to filter for these events at the API level in the get instead of in the transform